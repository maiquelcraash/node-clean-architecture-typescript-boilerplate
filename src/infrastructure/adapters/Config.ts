import { IAdapter } from '../../domain/ports/IAdapter.js';
import { IConfig } from '../../interface-adapters/ports/IConfig.js';

/**
 * Represents a configuration object.
 *
 * @class
 * @implements {IConfig}
 */
export class Config extends IConfig implements IAdapter {
    static SERVER_HOST = process.env.SERVER_HOST || 'localhost';
    static SERVER_PORT = Number(process.env.SERVER_PORT) || 3000;
    static SERVER_SSL = process.env.SERVER_SSL === 'true';
    static SERVER_SSL_CERT_PATH = undefined;
    static SERVER_SSL_PRIVATE_KEY_PATH = undefined;

    static BODY_SIZE_LIMIT = '50mb';
}
