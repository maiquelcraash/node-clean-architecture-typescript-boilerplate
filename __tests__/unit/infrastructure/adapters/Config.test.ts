import { Config } from '../../../../src/infrastructure/adapters/Config.js';

describe('infrastructure > adapters > Config', () => {
    afterEach(() => {
        jest.clearAllMocks();
        jest.restoreAllMocks();
    });

    describe('When importing config', () => {
        // just testing avoid accidental changing of config
        test('then should match snapshot', () => {
            expect({...Config}).toMatchSnapshot();
        });
    });
});
