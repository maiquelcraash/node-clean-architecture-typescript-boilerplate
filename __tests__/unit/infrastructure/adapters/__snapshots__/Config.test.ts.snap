// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infrastructure > adapters > Config When importing config then should match snapshot 1`] = `
{
  "BODY_SIZE_LIMIT": "50mb",
  "SERVER_HOST": "localhost",
  "SERVER_PORT": 3000,
  "SERVER_SSL": false,
  "SERVER_SSL_CERT_PATH": undefined,
  "SERVER_SSL_PRIVATE_KEY_PATH": undefined,
}
`;
